### PR TITLE
feat(project): replace defaults, auto-convert objects for x-ww-form-u…

### PR DESCRIPTION
**Pull request by doktordirk**
…rlencoded and json. allow form and multi-form data

BREAKING CHANGE: registerEndpoint - defaults will now completely replace standard deafults. Needed  to submit multipart/form-data.

---
This change is [Reviewable](https://reviewable.io/reviews/spoonx/aurelia-api/102)

### CODE REVIEW

File: src/rest.js [hunk 1]
```diff
@@ -33,12 +32,15 @@ export class Rest {
...
-    let requestOptions = extend(true, {}, this.defaults, options);
-
-    requestOptions.method = method;
-
-    if (typeof body === 'object') {
-      requestOptions.body = json(body);
+    let requestOptions = extend(true, {}, this.defaults, options, {method, body});
+
+    if (typeof body === 'object'
+    && !(typeof FormData === 'function' && body instanceof FormData)) {
```
Comment (RWOverdijk) [inline src/rest.js]: This indentation is not my liking. Combine the typeof and instanceof on body and put it all on one line. (You could _only_ use instance of, the typeof then becomes pointless)

File: src/rest.js [hunk 2]
```diff
@@ -33,12 +32,15 @@ export class Rest {
...
+      if (requestOptions.headers['Content-Type'].search('application/x-www-form-urlencoded') !== -1) {
```
Comment (RWOverdijk) [inline src/rest.js]: I'd use toLowerCase, and a strict equals. Written like:

```js
let contentType = requestOptions.headers['Content-Type'].toLowerCase();

 requestOptions.body = contentType === 'application/json' ? JSON.stringify(body) : qs.stringify(body);
```

**Comment by RWOverdijk** [comment]
👍

**Comment by doktordirk** [comment]
Reviewed 6 of 8 files at r1, 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/spoonx/aurelia-api/102)*

**Pull request merged**